opendatahub-io/opendatahub-operator
# Implement DSCInitialization Controller

Introduces changes defined in #245 

## Description
- [x] Ability to create Namespaces
- [x]  Ability to create namespace specific Network Policies, RoleBindings
- [x] Identifying if Addon Installation
- [x]  Ability to add Addon specific logic
- [x]  Optional Dev field to refer custom manifests 
- [ ]  Deploy Monitoring resources (Partial implementation, paused any changes until further confirmation regarding implementation of Monitoring stack)
- [x] Deploy Dashboard CRDs
- [ ]  Add Logs and Events
- [x] Cleanup resources 
- [x] Ability of Operator to create DSCInitialization CR
- [ ] Add Unit Tests for DSCInitialization Controller
- [ ]  Add ConsoleLink for Managed services

## How Has This Been Tested?

### Test already built image

1. Deploy Operator
```
 operator-sdk run bundle quay.io/vhire/opendatahub-operator-bundle:dev-0.0.1
```
2. Verify `default` DSCInitialization instance is created
3. Verify `opendatahub` namespace is created with networkpolicies and odh specific labels

### Test custom image build
1. Pull in updated manifests
```
make get-manifests -e MANIFEST_REPO=VaishnaviHire -e MANIFEST_RELEASE=rearch_manifests
```
2. Build operator Image
```
 make image -e IMG=<quay.io/username/opendatahub-operator:<TAG>>
```
4. Build Bundle
```
 make bundle-build bundle-push BUNDLE_IMG=quay.io/<username>/opendatahub-operator-bundle:<TAG>
```
5. Run Bundle
```
operator-sdk run bundle -e BUNDLE_IMG=quay.io/<username>/opendatahub-operator-bundle:<TAG>
```
## Issues
Fix Path for local manifest bundle


## Merge criteria:
